app/services/legacy/dataset_import_service.rb

```diff
@@ -35,49 +35,79 @@ def update_or_create_dataset
     dataset.location1 = build_location
     dataset.licence = build_licence
     dataset.licence_other = build_licence_other
-    old_theme  = legacy_dataset["theme-primary"]
-    secondary_theme  = legacy_dataset["theme-secondary"]
+    old_theme = legacy_dataset["theme-primary"]
+    secondary_theme = legacy_dataset["theme-secondary"]
     dataset.theme_id = themes_cache.fetch(old_theme, nil)
     dataset.secondary_theme_id = themes_cache.fetch(secondary_theme, nil)
     dataset.status = "published"
     dataset.save!(validate: false)
   end
 
   def create_datafiles(dataset)
```

Comment:
Not important now, but recommend for a future PR ... the database_importer_service has a lot of public instance methods which should either be private (the private methods should be tested by calling the public one, which in this case will probably be #run) or could be moved to public interfaces on other classes.

Reply:
yeah agreed. This PR is already massive, so let's do that in another PR. 👍